provider-announcements/src/java/org/sakaiproject/archiver/provider/AnnouncementsArchiver.java

```diff
@@ -0,0 +1,111 @@
+import org.sakaiproject.announcement.api.AnnouncementService;
+import org.sakaiproject.archiver.api.Archiveable;
+import org.sakaiproject.archiver.api.ArchiverRegistry;
+import org.sakaiproject.archiver.api.ArchiverService;
+import org.sakaiproject.archiver.util.Jsonifier;
+import org.sakaiproject.content.api.ContentHostingService;
+import org.sakaiproject.entity.api.Reference;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.exception.ServerOverloadException;
+import org.sakaiproject.exception.TypeException;
+import org.sakaiproject.message.api.Message;
+import org.sakaiproject.site.api.SiteService;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Implementation of {@link Archiveable} for the announcements tool
+ */
+
+@Slf4j
+public class AnnouncementsArchiver implements Archiveable {
+
+	private static final String ANNOUNCEMENTS_TOOL = "sakai.announcements";
+
+	public void init() {
+		ArchiverRegistry.getInstance().register(ANNOUNCEMENTS_TOOL, this);
+	}
+
+	public void destroy() {
+		ArchiverRegistry.getInstance().unregister(ANNOUNCEMENTS_TOOL);
+	}
+
+	@Setter
+	private ArchiverService archiverService;
+	
+	@Setter 
+	private AnnouncementService announcementService;
+	
+	@Setter
+	private ContentHostingService contentHostingService;
+
+	@Override
+	public void archive(final String archiveId, final String siteId, final String toolId, final boolean includeStudentContent) {
+		
+		try {
+			
+			// Get the announcements for this site
+			String channelRef = this.announcementService.channelReference(siteId, SiteService.MAIN_CONTAINER);
+			List<Message> announcements = new ArrayList<Message>();
+			announcements.addAll(this.announcementService.getMessages(channelRef, null, 0, true, false, false));
+			
+			// Go through each announcement and save the data we are interested in
+			for (Message message : announcements) {
+				AnnouncementMessage announcement = (AnnouncementMessage) message;
+				
+				// Save the metadata for this announcement
+				ArchiveItem archiveItem = new ArchiveItem();
+				archiveItem.setTitle(announcement.getAnnouncementHeader().getSubject());
+				archiveItem.setCreatedBy(announcement.getHeader().getFrom().getDisplayName());
+				archiveItem.setCreatedOn(new Date(announcement.getHeader().getDate().getTime()));
+				String jsonArchiveItem = Jsonifier.toJson(archiveItem);
+				log.debug("Announcement metadata: " + jsonArchiveItem);
+				this.archiverService.archiveContent(archiveId, siteId, toolId, jsonArchiveItem.getBytes(), archiveItem.getTitle() + " (metadata).json");
+				
+				// Save the body of the announcement
+				String announcementBody = announcement.getBody();
+				log.debug("Announcement body: " + announcementBody);
+				this.archiverService.archiveContent(archiveId, siteId, toolId, announcementBody.getBytes(), archiveItem.getTitle() + " (content).html");
```

Comment:
Is this separate because it is HTML? It is weird having the metadata and content of an announcement separate. I wonder if its possible to tell Jsonifier to not convert the HTML?